### PR TITLE
Added Event Emitter and Callstats Service

### PR DESCRIPTION
**Pull request by code-master5**
This PR adds:

    1.  Event emitter support to Jangouts 
I've used RxJS (thanks to @imobachgs for suggestion). The `events.provider.js` is a module that 
provides an interface in form of a `Subject` to which any external service can subscribe to receive 
events. These events, then, can be used by services according to their use cases.

    2.  Callstats Service Integration
I've added `callstats.provider.js` as a module that reads callstats user configuration from 
`callstats.config.json` file. It subscribes to the event emitter and recieves events, which, it then
sends to the callstats.io service.

### CODE REVIEW

**Comment by code-master5** [comment]
@ancorgs Please review this.

File: src/app/components/feed/feeds.factory.js [hunk 1]
```diff
@@ -290,6 +292,26 @@
                 if (options.after) { options.after(); }
                 // Send the new status to remote peers
                 DataChannelService.sendStatus(that, {exclude: "picture"});
+
```
Comment (ancorgs) [inline src/app/components/feed/feeds.factory.js]: I would make this event more generic. Something like (just a possible example)
```
type: "channel",
data {
  channel: type,
  status: enabled,
  peerconnection: that.connection.pluginHandle.webrtcStuff.pc
}
```

The fact that video and audio are considered different (for example "paused" vs "muted") is very specific to callstats.io, not a real general difference for Jangouts. So that "if" should be in the callstats provider.

File: src/app/components/room/room.service.js [hunk 2]
```diff
@@ -353,10 +428,27 @@
       var display = FeedsService.findMain().display;
       var connection;
       var id;
-
+      
+      // emit `screenshare` event 
+      jhEventsProvider.emitEvent({
+        type: "screenshare",
```
Comment (ancorgs) [inline src/app/components/room/room.service.js]: This event also looks very influenced by callstats API. I don't think we need a different event for publishing screensharing and for publishing camera, just a different argument to the same type of event. From the jangouts point of view both are basically the same event even if callstats has a dedicated screenshare start/stop events.

File: src/app/components/room/room.service.js [hunk 3]
```diff
@@ -106,10 +106,33 @@
       var $$rootScope = $rootScope;
       var connection = null;
 
+      // adding username to jhEventsProvider
+      jhEventsProvider.username = username;
+      // adding room details to jhEventsProvider
+      jhEventsProvider.roomDesc = that.room.description;
```
Comment (ancorgs) [inline src/app/components/room/room.service.js]: I wouldn't initialize this here. I would make the callstats provider ask directly the information it needs to `UserService` and `RoomService`. This is another unnecessary coupling of our event emitter with the callstats take on things.

File: src/app/callstats/callstats.config.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "AppID": "724446896",
+  "AppSecret": "/BKw/++t+DXe:UCxvwuXdkE6AIuxHXRj5PnaKLNeAgNZdp6rGShdWNmc="
+}
```
Comment (ancorgs) [inline src/app/callstats/callstats.config.json]: This should not be part of the PR, of course.

File: src/app/components/feed/feeds.factory.js (hunk 1 above)
Comment (code-master5) [inline src/app/components/feed/feeds.factory.js]: @ancorgs Done! Please check this.

File: src/app/components/room/room.service.js (hunk 3 above)
Comment (ancorgs) [reply src/app/components/room/room.service.js]: @code-master5 you told me about this being not possible. But I did some experiments with the logic of `jsEventsProvider` and `Callstats` being placed in new components within the `janusHangouts` module (instead of defining their own modules) and I don't see why shouldn't it work.

Comment (code-master5) [reply src/app/components/room/room.service.js]: @ancorgs Sorry, I can't understand. Can you provide me the link to your experiments for reference? Things were not working in my case! May be I missed something here.

File: src/app/callstats/callstats.config.json (hunk 4 above)
Comment (code-master5) [reply src/app/callstats/callstats.config.json]: @ancorgs Done!

**Comment by code-master5** [comment]
@ancorgs Hey there! nice to hear from you, after so long. Finally, this is merged and I'm happy. I'm very thankful to you and @imobachgs for constant help and mentoring. It was fun working with you guys.
